allennlp/common/file_utils.py

```diff
@@ -325,12 +329,24 @@ def cached_path(
 
     if extraction_path is not None:
         # If the extracted directory already exists (and is non-empty), then no
-        # need to extract again unless `force_extract=True`.
+        # need to create a lock file and extract again unless `force_extract=True`.
         if os.path.isdir(extraction_path) and os.listdir(extraction_path) and not force_extract:
             return extraction_path
 
         # Extract it.
         with FileLock(extraction_path + ".lock"):
```

Comment:
This will make sure that only one process/thread makes it into the block below, right?

Reply:
yeup